src/network/networkd-routing-policy-rule.c

```diff
@@ -1579,6 +1610,12 @@ static int routing_policy_rule_section_verify(RoutingPolicyRule *rule) {
                                 "specified by To= or From=. Ignoring [RoutingPolicyRule] section from line %u.",
                                 rule->section->filename, rule->section->line);
 
+        if ((rule->family == AF_INET6) && (rule->from_dhcp_ip || rule->to_dhcp_ip || rule->from_dhcp_gateway || rule->to_dhcp_gateway))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "%s: address family specified by Family= conflicts with the special DHCP4 values "
+                                       "specified by To= or From=. Ignoring [RoutingPolicyRule] section from line %u.",
+                                       rule->section->filename, rule->section->line);
+
```

Comment:
Please move this check after the `if (rule->family == AF_UNSPEC)` block below, as `rule->family` may be updated in the block.

Also, not only `rule->family == AF_INET6`, but also check `FLAGS_SET(rule->address_family, ADDRESS_FAMILY_IPV6`. That is,
```
if ((rule->family == AF_INET6 || FLAGS_SET(rule->address_family, ADDRESS_FAMILY_IPV6)) &&
    (rule->from_type != ADDRESS_TYPE_STATIC || rule->to_type != ADDRESS_TYPE_STATIC))
        return log_error_errno(...);
```